cli/diag.c

```diff
@@ -0,0 +1,317 @@
+		printf("%4d  %7d      %8d\n", i, coeff.cursors[i].pre,
+		       coeff.cursors[i].post);
+	}
+
+	return 0;
+}
+
+#define CMD_DESC_PORT_EQ_TXFSLF "Dump FS/LF output data"
+
+static int port_eq_txfslf(int argc, char **argv)
+{
+	struct diag_common_cfg cfg = DEFAULT_DIAG_COMMON_CFG;
+	struct switchtec_port_eq_tx_fslf data;
+	int i, ret;
+
+	const struct argconfig_options opts[] = {
+		DEVICE_OPTION, FAR_END_OPTION, PORT_OPTION, PREV_OPTION, {}
+	};
+
+	ret = diag_parse_common_cfg(argc, argv, CMD_DESC_PORT_EQ_TXFSLF,
+				    &cfg, opts);
+	if (ret)
+		return ret;
+
+	printf("%s Equalization FS/LF data for physical port %d %s\n\n",
+	       cfg.far_end ? "Far End" : "Local", cfg.port_id,
+	       cfg.prev ? "(Previous Link-Up)" : "");
+	printf("Lane    FS    LF\n");
+
+	for (i = 0; i < cfg.port.neg_lnk_width; i++) {
+		ret = switchtec_diag_port_eq_tx_fslf(cfg.dev, cfg.port_id, i,
+				cfg.end, cfg.link, &data);
+		if (ret) {
+			switchtec_perror("port_eq_fs_ls");
+			return -1;
+		}
+
+		printf("%4d  %4d  %4d\n", i, data.fs, data.lf);
+	}
+
+	return 0;
+}
+
+#define CMD_DESC_PORT_EQ_TXTABLE "Dump far end port equalization table"
+
+static int port_eq_txtable(int argc, char **argv)
+{
+	struct diag_common_cfg cfg = DEFAULT_DIAG_COMMON_CFG;
+	struct switchtec_port_eq_table table;
+	int i, ret;
+
+	const struct argconfig_options opts[] = {
+		DEVICE_OPTION, PORT_OPTION, PREV_OPTION, {}
+	};
+
+	ret = diag_parse_common_cfg(argc, argv, CMD_DESC_PORT_EQ_TXTABLE,
+				    &cfg, opts);
+	if (ret)
+		return ret;
+
+	ret = switchtec_diag_port_eq_tx_table(cfg.dev, cfg.port_id,
+					      cfg.link, &table);
+	if (ret) {
+		switchtec_perror("port_eq_table");
+		return -1;
+	}
+
+	printf("Far End TX Equalization Table for physical port %d, lane %d %s\n\n",
+	       cfg.port_id, table.lane_id, cfg.prev ? "(Previous Link-Up)" : "");
+	printf("Step  Pre-Cursor  Post-Cursor  FOM  Pre-Up  Post-Up  Error  Active  Speed\n");
+
+	for (i = 0; i < table.step_cnt; i++) {
+		printf("%4d  %10d  %11d  %3d  %6d  %7d  %5d  %6d  %5d\n",
+		       i, table.steps[i].pre_cursor, table.steps[i].post_cursor,
+		       table.steps[i].fom, table.steps[i].pre_cursor_up,
+		       table.steps[i].post_cursor_up, table.steps[i].error_status,
+		       table.steps[i].active_status, table.steps[i].speed);
+	}
+
+	return 0;
+}
+
+#define CMD_DESC_RCVR_OBJ "Dump analog RX coefficients/adaptation objects"
+
+static int rcvr_obj(int argc, char **argv)
+{
+	struct diag_common_cfg cfg = DEFAULT_DIAG_COMMON_CFG;
+	struct switchtec_rcvr_obj obj;
+	int i, j, ret;
+
+	const struct argconfig_options opts[] = {
+		DEVICE_OPTION, PORT_OPTION, PREV_OPTION, {}
+	};
+
+	ret = diag_parse_common_cfg(argc, argv, CMD_DESC_RCVR_OBJ,
+				    &cfg, opts);
+	if (ret)
+		return ret;
+
+	printf("Coefficients for physical port %d %s\n\n", cfg.port_id,
+	       cfg.prev ? "(Previous Link-Up)" : "");
+	printf("Lane  CTLE  Tgt Amp  Spec DFE  DFE0 DFE1 DFE2 DFE3 DFE4 DFE5 DFE6\n");
```

Comment:
The 'Tgt Amp' and 'Spec DFE' seems like 4 column header in the output. Use a '-' to connect them?